utils/frame/benchmarking-cli/src/writer.rs

```diff
@@ -130,69 +100,120 @@ pub fn write_results(file: &mut File, batches: Vec<BenchmarkBatch>) -> Result<()
 
 		// only create new trait definitions when we go to a new pallet
 		if batch.pallet != current_pallet {
-			if !current_pallet.is_empty() {
-				// close trait
-				write!(file, "}}\n").unwrap();
-			}
+			// auto-generation note
+			write!(
+				file,
+				"//! THIS FILE WAS AUTO-GENERATED USING THE SUBSTRATE BENCHMARK CLI VERSION {}\n\n",
+				 VERSION,
+			)?;
+
+			// general imports
+			write!(
+				file,
+				"use frame_support::weights::{{Weight, constants::RocksDbWeight as DbWeight}};\n\n"
+			)?;
 
 			// struct for weights
-			write!(file, "pub struct WeightFor{};\n",
-				pallet_string.to_pascal_case(),
-			).unwrap();
+			write!(file, "pub struct WeightInfo;\n")?;
 
 			// trait wrapper
-			write!(file, "impl {}::WeightInfo for WeightFor{} {{\n",
-				pallet_string,
-				pallet_string.to_pascal_case(),
-			).unwrap();
+			write!(file, "impl {}::WeightInfo for WeightInfo {{\n", pallet_string)?;
 
 			current_pallet = batch.pallet.clone()
 		}
 
-		// function name
-		write!(file, "\tfn {}(", benchmark_string).unwrap();
+		// Analysis results
+		let extrinsic_time = Analysis::min_squares_iqr(&batch.results, BenchmarkSelector::ExtrinsicTime).unwrap();
```

Comment:
This is a different error type than what is outputted from the FileIO. Would need to do some more fancy error management, which I tried but didnt lead to anything better tbh

Reply:
If you know how to better handle multiple error types in this scenario, would def appreciate a commit that fixes it.

Reply:
I'll give it a quick try.

Reply:
My impression: the issue with the errors seems to be that the benchmarking command uses sc_cli::Error instead of defining its own, which restricts which errors you can bubble up.
so we'd have to adjust that in order to integrate custom errors.
--> out of scope, maybe address later

Reply:
I guess min_square_iqr does not return an io error, thus it require a bit more refactor